src/Reports/index.ts

```diff
@@ -6,81 +6,99 @@ import {
   trancheSnapshotsPostProcess,
   trancheSnapshotsQuery,
 } from '../queries/trancheSnapshots.js'
-import { processBalanceSheetData } from './processors/balanceSheet.js'
 import { combineLatest, of } from 'rxjs'
-import { ReportProcessorType } from './processors/index.js'
+import { processor } from './Processor.js'
 
 import { map } from 'rxjs'
-import { GroupBy } from '../utils/date.js'
-export interface ReportFilter {
-  from?: string
-  to?: string
-  groupBy?: GroupBy
-}
+import { BalanceSheetReport, CashflowReport, ReportFilter } from './types.js'
+import { Query } from '../types/query.js'
+import {
+  PoolFeeSnapshotFilter,
+  poolFeeSnapshotQuery,
+  poolFeeSnapshotsPostProcess,
+} from '../queries/poolFeeSnapshots.js'
+import { PoolMetadata } from '../types/poolMetadata.js'
 
-export interface ReportData {
-  timestamp: string
-  [key: string]: unknown
-}
+type ReportType = 'balanceSheet' | 'cashflow'
 
 export class Reports extends Entity {
   constructor(
     centrifuge: Centrifuge,
-    public poolId: string
+    public poolId: string,
```

Comment:
both of these can be replaced with `public pool: Pool`. then line 53 can be `const metadata$ = this.pool.metadata()`